src/test/scala/com/stratio/hermes/utils/HermesTest.scala

```diff
@@ -135,6 +135,41 @@ class HermesTest extends FlatSpec with Matchers {
     //scalastyle:on
   }
 
+  it should "generate valid locations: ES and US locales" in {
+
+    val hermesES = Hermes("ES")
+    hermesES.Geo.geolocationOfGeoModel(hermesES.Geo.geoModel) should contain (hermesES.Geo.geolocation)
+
+    val hermesUS = Hermes("US")
+    hermesUS.Geo.geolocationOfGeoModel(hermesUS.Geo.geoModel) should contain (hermesUS.Geo.geolocation)
+  }
+
+  it should "raise a NoSuchElementException when the locale is empty" in {
+    val hermes = Hermes("XX")
+    //scalastyle:off
+    an[NoSuchElementException] should be thrownBy hermes.Geo.geolocation
+    //scalastyle:on
+  }
+
+  it should "when you do not specify the locale try to use all the locales" in {
+    val hermes = Hermes()
+    hermes.Geo.geolocationOfGeoModel(hermes.Geo.geoModel) should contain (hermes.Geo.geolocation)
+  }
+
+  it should "raise an exception when it gets a geolocation that not exists" in {
+    val hermesFR = Hermes("FR")
+    //scalastyle:off
+    an[NoSuchElementException] should be thrownBy hermesFR.Geo.geolocation
+    //scalastyle:on
+  }
+
+  it should "raise an exception when it gets a geolocation that is corrupted" in {
+    val hermesYY = Hermes("YY")
+    //scalastyle:off
+    hermesYY.Geo.parseErrorList(hermesYY.Geo.geoModel).length should be(1)
+    an[NoSuchElementException] should be thrownBy hermesYY.Geo.geolocation
+    //scalastyle:on
+  }
   /**
```

Comment:
With the changes that you've just made I guess we should include more tests to fulfill the acceptance criteria.